gazelle/bzl/gazelle.go

```diff
@@ -324,3 +322,20 @@ func (s srcsList) Contains(m string) bool {
 	}
 	return false
 }
+
+// checkInternalVisibility overrides the given visibility if the package is
+// internal. This is taken from Gazelle's rule package. Unfortunately bzl
+// allows for both the name "internal" and "private" to indicate visibility and
+// so it can't be used verbatim, but with slight modifications it can be.
+// https://github.com/bazelbuild/bazel-gazelle/blob/2078a44d2bba287db98b2b48b5f6cf4f08642489/rule/rule.go#L833:6
+func checkInternalVisibility(rel, visibility string) string {
+	rel = rel + "/" // Append a / to make finding the suffix easier.
```

Comment:
Maybe use `github.com/bazelbuild/bazel-gazelle/pathtools.Index` instead? It's like `strings.Index`, but it only matches on path component boundaries.

Reply:
I made the change and it is passing tests so I believe it is equivalent. The only reason I didn't use `pathtools.Index` instead from the start is because it wasn't looking for the `LastIndex` and I didn't want it to turn `/foo/internal/bar/internal` into being visible to `//foo:__subpackages__`